### ISSUE TITLE
Renew of Let's Encrypt cert fails due to "Request is not json" Error

### ISSUE THREAD

**Issue by saschaludwig** [traefik/traefik#10153]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

cyclical renew and issue of new certs fails

### What did you see instead?

See the below attached log excerpt.
I see "Request is not json" erros in the log output

### What version of Traefik are you using?

```
Version:      3.0.0-beta3
Codename:     beaufort
Go version:   go1.20.5
Built:        2023-06-22T08:58:13Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration?

```yaml
version: '3'

services:
  reverse-proxy:
    image: traefik:v3.0.0-beta3
    restart: always
    command:
      - "--api.insecure=true"
      - "--providers.docker"
      - "--certificatesresolvers.letsencrypt.acme.caserver=https://acme-v02.api.letsencrypt.org/directory"
      - "--certificatesresolvers.letsencrypt.acme.email=user@somedomain"
      - "--certificatesresolvers.letsencrypt.acme.storage=/traefik-certificates-letsencrypt/acme.json"
      - "--certificatesresolvers.letsencrypt.acme.dnschallenge=true"
      - "--certificatesresolvers.letsencrypt.acme.dnschallenge.provider=pdns"
      - "--entrypoints.http.address=:80"
      - "--entrypoints.https.address=:443"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
    environment:
      - "PDNS_API_URL=https://myacmednsserver"
      - "PDNS_API_KEY=myfancyapikey"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - traefik-certificates-letsencrypt:/traefik-certificates-letsencrypt
      
    networks:
      proxy:
      traefik:
        ipv4_address: x.x.x.x
        
networks:
  traefik:
    name: macvlan network
    external: true
  proxy:
    external: true


volumes:
  traefik-certificates-letsencrypt:

```

Add more configuration information here.

### If applicable, please paste the log output in DEBUG level

```
2023-10-05T16:45:48Z INF github.com/traefik/traefik/v3/pkg/provider/acme/provider.go:820 > Renewing certificate from LE : {Main:evcc.home.somewhere.com SANs:[]} acmeCA=https://acme-v02.api.letsencrypt.org/directory providerName=letsencrypt.acme
2023-10-05T16:45:48Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Trying renewal with 248 hours remaining lib=lego
2023-10-05T16:45:48Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Obtaining bundled SAN certificate lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/271016355726 lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Could not find solver for: tls-alpn-01 lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Could not find solver for: http-01 lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: use dns-01 solver lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Preparing to solve DNS-01 lib=lego
2023-10-05T16:45:49Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] [evcc.home.somewhere.com] acme: Cleaning DNS-01 challenge lib=lego
2023-10-05T16:45:50Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [WARN] [evcc.home.somewhere.com] acme: cleaning up failed: unexpected status code: [status code: 400] body: {"msg":"Request is not json"}  lib=lego
2023-10-05T16:45:50Z DBG github.com/go-acme/lego/v4@v4.12.2/log/logger.go:48 > [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/271016355726 lib=lego
2023-10-05T16:45:50Z ERR github.com/traefik/traefik/v3/pkg/provider/acme/provider.go:828 > Error renewing certificate from LE: {evcc.home.somewhere.com []} error="error: one or more domains had a problem:\n[evcc.home.somewhere.com] [evcc.home.somewhere.com] acme: error presenting token: unexpected status code: [status code: 400] body: {\"msg\":\"Request is not json\"}\n" acmeCA=https://acme-v02.api.letsencrypt.org/directory providerName=letsencrypt.acme
```

**Comment by saschaludwig** [comment]
That problem is also present in the latest stable release:
```
Version:      2.10.4
Codename:     saintmarcelin
Go version:   go1.20.6
Built:        2023-07-24T16:29:02Z
OS/Arch:      linux/amd64
```

**Comment by ldez** [comment]
Hello,

the problem is already fixed in #10087 with the update to lego v4.14.0

v3.0.0-beta3 and v2.10.4 use the previous version of lego:
- https://github.com/traefik/traefik/blob/v3.0.0-beta3/go.mod#L22 (v4.12.2)
- https://github.com/traefik/traefik/blob/v2.10.4/go.mod#L22 (v4.13.2)

The fix will be available in the next releases of Traefik.
